src/Kafka/Consumer/ConsumerProperties.hs

```diff
@@ -123,3 +127,13 @@ queuedMaxMessagesKBytes :: Int -> ConsumerProperties
 queuedMaxMessagesKBytes kBytes =
   extraProp "queued.max.messages.kbytes" (Text.pack $ show kBytes)
 {-# INLINE queuedMaxMessagesKBytes #-}
+
+-- | The user will poll the consumer frequently to handle both new
+-- messages and rebalance events.
+--
+-- By default hw-kafka-client handles polling rebalance events for you
+-- in a background thread, with this property set you can simplify
+-- hw-kafka-client's footprint and have full control over when polling
+-- happens at the cost of having to manage this yourself.
+userPolls :: ConsumerProperties
```

Comment:
The name `userPolls` kind of suggests that user should somehow poll manually.

Should we instead do something like:
```
data CallbackMode = CallbackModeAsync | CallbackModeSync

callbackMode :: CallbackMode -> ConsumerProperties
```

It looks a bit more explicit to me, do you agree?